### ISSUE TITLE
'Error binding parameter 6 - probably unsupported type' while importing

### ISSUE THREAD

**Issue by janpeeters70** [beetbox/beets#3473]
Hi I'm getting the above error while importing the album 
Fantomas - Fantomas

I've run a verbose import and that gives the following output

````
beet -vv import /Volumes/Stor/Music\ Torrents/Fantômas\ -\ Fantômas\ \(1998\)\ \(FLAC\)
user configuration: /Users/jpeeters/.config/beets/config.yaml
data directory: /Users/jpeeters/.config/beets
plugin paths: 
Sending event: pluginload
inline: adding item field multidisc
inline: adding item field my_samplerate
inline: adding item field is_flac
inline: adding album field format
inline: adding album field av_bitrate
inline: adding album field album_bitdepth
inline: adding album field album_samplerate
library database: /Users/jpeeters/Music/Beets/musiclibrary.db
library directory: /Volumes/Tunez
Sending event: library_opened
Sending event: import_begin
Sending event: import_task_created
Sending event: import_task_start
Looking up: /Volumes/Stor/Music Torrents/Fantômas - Fantômas (1998) (FLAC)
Tagging Fantômas - Fantômas
No album ID found.
Search terms: Fantômas - Fantômas
Album might be VA: False
Searching for MusicBrainz releases with: {'release': 'fantômas', 'artist': 'fantômas', 'tracks': '30'}
Requesting MusicBrainz release 64080ec9-80db-34f2-89be-ba638e41a5a7
primary MB release type: album
Sending event: albuminfo_received
Candidate: Fantômas - Fantômas (64080ec9-80db-34f2-89be-ba638e41a5a7)
Computing track assignment...
...done.
Success. Distance: 0.27
Requesting MusicBrainz release c04eff45-5240-4ada-9334-a3a5784f6ad3
primary MB release type: album
Sending event: albuminfo_received
Candidate: Fantômas - Fantômas (c04eff45-5240-4ada-9334-a3a5784f6ad3)
Computing track assignment...
...done.
Success. Distance: 0.27
Requesting MusicBrainz release f5c2979d-4b54-4529-8fc3-330af8332dac
primary MB release type: album
Sending event: albuminfo_received
Candidate: Formacja Nieżywych Schabuff - Fantomas (f5c2979d-4b54-4529-8fc3-330af8332dac)
Computing track assignment...
...done.
Success. Distance: 0.71
Requesting MusicBrainz release 09845e85-24e9-488d-b5ca-1ea13b450f1b
primary MB release type: album
secondary MB release type(s): compilation
Sending event: albuminfo_received
Candidate: Michel Magne - Fantômas (09845e85-24e9-488d-b5ca-1ea13b450f1b)
Computing track assignment...
...done.
Success. Distance: 0.62
Requesting MusicBrainz release 03716936-9636-4db8-9999-5c806a33075f
primary MB release type: single
Sending event: albuminfo_received
Candidate: Demon Eyes - Fantomas (03716936-9636-4db8-9999-5c806a33075f)
Computing track assignment...
...done.
Success. Distance: 0.77
chroma: acoustid album candidates: 0
spotify: Searching Spotify for 'album:Fantomas artist:Fantomas'
spotify: Found 1 results from Spotify for 'album:Fantomas artist:Fantomas'
spotify: Searching for album 01BbebmwbMQM5njECmgDIV
Sending event: albuminfo_received
Candidate: Fantomas - Fantomas (01BbebmwbMQM5njECmgDIV)
Computing track assignment...
...done.
Success. Distance: 0.19
discogs: Searching for master release 168408
discogs: hit rate limit, waiting for 1.8081909656524657 seconds
discogs: Searching for master release 114303
discogs: hit rate limit, waiting for 1.1542810440063476 seconds
discogs: Searching for master release 114303
discogs: hit rate limit, waiting for 1.6817399501800536 seconds
Sending event: albuminfo_received
Candidate: Mastik Soul - Mastik Fantastik EP (1267229)
Computing track assignment...
...done.
Success. Distance: 0.84
Sending event: albuminfo_received
Candidate: Pixies - Lynn. Mass. A Fantastic Live Act (3498135)
Computing track assignment...
...done.
Success. Distance: 0.70
Sending event: albuminfo_received
Candidate: Leopold Stokowski And The Philadelphia Orchestra - Walt Disney's Masterpiece Fantasia (6259607)
Computing track assignment...
...done.
Success. Distance: 0.75
Sending event: albuminfo_received
Candidate: Leopold Stokowski And The Philadelphia Orchestra - Walt Disney's Masterpiece Fantasia (4472182)
Computing track assignment...
...done.
Success. Distance: 0.75
Sending event: albuminfo_received
Candidate: Outcast - Mass Murder Fantasies (5164383)
Computing track assignment...
...done.
Success. Distance: 0.70
Evaluating 11 candidates.

/Volumes/Stor/Music Torrents/Fantômas - Fantômas (1998) (FLAC) (30 items)
Sending event: before_choose_candidate
Correcting tags from:
    Fantômas - Fantômas
To:
    Fantomas - Fantomas
URL:
    https://open.spotify.com/album/01BbebmwbMQM5njECmgDIV
(Similarity: 81.1%) (tracks, source, year) (Spotify, [1999], Ipecac Recordings)
 * Page 1 (6 Frames)   -> Page 1 (source, title)
 * Page 2 (7 Frames)   -> Page 2 (source, title)
 * Page 3 (17 Frames)  -> Page 3 (source, title)
 * Page 4 (11 Frames)  -> Page 4 (source, title)
 * Page 5 (7 Frames)   -> Page 5 (source, title)
 * Page 6 (26 Frames)  -> Page 6 (source, title)
 * Page 7 (6 Frames)   -> Page 7 (source, title)
 * Page 8 (9 Frames)   -> Page 8 (source, title)
 * Page 9 (11 Frames)  -> Page 9 (source, title)
 * Page 10 (15 Frames) -> Page 10 (source, title)
 * Page 11 (10 Frames) -> Page 11 (source, title)
 * Page 12 (31 Frames) -> Page 12 (source, title)
 * Page 13             -> Page 13 (source)
 * Page 14 (4 Frames)  -> Page 14 (source, title)
 * Page 15 (22 Frames) -> Page 15 (source, title)
 * Page 16 (11 Frames) -> Page 16 (source, title)
 * Page 17 (14 Frames) -> Page 17 (source, title)
 * Page 18 (20 Frames) -> Page 18 (source, title)
 * Page 19 (21 Frames) -> Page 19 (source, title)
 * Page 20 (13 Frames) -> Page 20 (source, title)
 * Page 21 (11 Frames) -> Page 21 (source, title)
 * Page 22 (7 Frames)  -> Page 22 (source, title)
 * Page 23 (17 Frames) -> Page 23 (source, title)
 * Page 24 (19 Frames) -> Page 24 (source, title)
 * Page 25 (34 Frames) -> Page 25 (source, title)
 * Page 26 (7 Frames)  -> Page 26 (source, title)
 * Page 27 (15 Frames) -> Page 27 (source, title)
 * Page 28 (20 Frames) -> Page 28 (source, title)
 * Page 29 (39 Frames) -> Page 29 (source, title)
 * Page 30 (2 Frames)  -> Page 30 (source, title)
[A]pply, More candidates, Skip, Use as-is, as Tracks, Group albums,
Enter search, enter Id, aBort, eDit, edit Candidates? 
Sending event: import_task_choice
Sending event: import_task_apply
0 of 30 items replaced
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/beet", line 11, in <module>
    load_entry_point('beets==1.4.9', 'console_scripts', 'beet')()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/ui/__init__.py", line 1266, in main
    _raw_main(args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/ui/__init__.py", line 1253, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/ui/commands.py", line 955, in import_func
    import_files(lib, paths, query)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/ui/commands.py", line 925, in import_files
    session.run()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/importer.py", line 329, in run
    pl.run_parallel(QUEUE_SIZE)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/util/pipeline.py", line 445, in run_parallel
    six.reraise(exc_info[0], exc_info[1], exc_info[2])
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/six.py", line 693, in reraise
    raise value
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/util/pipeline.py", line 312, in run
    out = self.coro.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/util/pipeline.py", line 171, in coro
    task = func(*(args + (task,)))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/importer.py", line 1420, in user_query
    apply_choice(session, task)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/importer.py", line 1491, in apply_choice
    task.add(session.lib)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/importer.py", line 756, in add
    self.album = lib.add_album(self.imported_items())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/library.py", line 1350, in add_album
    album.add(self)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/library.py", line 355, in add
    super(LibModel, self).add(lib)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/dbcore/db.py", line 581, in add
    self.store()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/library.py", line 1210, in store
    super(Album, self).store(fields)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/library.py", line 347, in store
    super(LibModel, self).store(fields)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/dbcore/db.py", line 510, in store
    tx.mutate(query, subvars)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/beets/dbcore/db.py", line 832, in mutate
    cursor = self.db._connection().execute(statement, subvals)
sqlite3.InterfaceError: Error binding parameter 6 - probably unsupported type.
````

The contents of the folder looks like this:
````
22 Page 22 (7 Frames).flac
06 Page 6 (26 Frames).flac
04 Page 4 (11 Frames).flac
12 Page 12 (31 Frames).flac
28 Page 28 (20 Frames).flac
02 Page 2 (7 Frames).flac
15 Page 15 (22 Frames).flac
03 Page 3 (17 Frames).flac
18 Page 18 (20 Frames).flac
26 Page 26 (7 Frames).flac
25 Page 25 (34 Frames).flac
01 Page 1 (6 Frames).flac
16 Page 16 (11 Frames).flac
19 Page 19 (21 Frames).flac
07 Page 7 (6 Frames).flac
30 Page 30 (2 Frames).flac
11 Page 11 (10 Frames).flac
21 Page 21 (11 Frames).flac
29 Page 29 (39 Frames).flac
23 Page 23 (17 Frames).flac
10 Page 10 (15 Frames).flac
20 Page 20 (13 Frames).flac
14 Page 14 (4 Frames).flac
27 Page 27 (15 Frames).flac
08 Page 8 (9 Frames).flac
09 Page 9 (11 Frames).flac
24 Page 24 (19 Frames).flac
05 Page 5 (7 Frames).flac
17 Page 17 (14 Frames).flac
13 Page 13.flac
Fantômas.cue
Fantômas - Fantômas.m3u
Fantômas - Fantômas.log
Fantômas - Fantômas.jpg
````

I'm on macOS 10.13.6

**Comment by sampsyo** [comment]
Huh! I'm pretty stumped. I don't see anything in this traceback that makes it obvious what's going wrong.

I notice this was tagged with Spotify. Does this go away if you disable the Spotify plugin?

**Comment by janpeeters70** [comment]
Yes indeed, deactivating the Spotify plugin made the import go through. 
Question remains why does the Spotify plugin obstruct the import?
I've only got the following code in my config.yaml.
As far as I know I don't ask it to try to tag things.

````
spotify:
    mode: open
    show_failures: on
    tiebreak: first
````

**Comment by Julian** [comment]
In my case after some debugging, it looks like the issue is essentially this line is likely misparsing the spotify API response:

https://github.com/beetbox/beets/blob/31855a93949a7b2e2636f709c2678d0efef1dcba/beetsplug/spotify.py#L178

Specifically for my album, spotify has only a year-granular precision, but the line doing the (vacuous) splitting will turn `1996` into `[1996]` and then try and insert that list into the DB instead of the int.

Seems likely the fix is to not do the split at all if the precision is yearly (or to check whether there's a `-` in the response maybe), but I don't really know the Spotify API's behavior well enough, or whether this will break other cases.

Happy to send a PR though if that sounds reasonable, it does certainly fix my own case.

**Comment by sampsyo** [comment]
Hmm… I'm not quite sure I see it. In the following lines, the list gets split up into its component parts, and `date_parts` itself goes unused thereafter:
https://github.com/beetbox/beets/blob/31855a93949a7b2e2636f709c2678d0efef1dcba/beetsplug/spotify.py#L183-L191

Doesn't that mean that each of `year`, `month`, and `day` is a single number? (And that's what we put in the database.) Maybe I'm missing something?

**Comment by Julian** [comment]
Ah, apologies I didn't notice that that line on master is different than what I have locally (line 189) -- looks like someone fixed this already then: https://github.com/beetbox/beets/commit/7ec363230903ebc7dfcbf0660efb3be78c2e63fd

Seems like beets might just need a release then? Installing beets (in e.g. a fresh venv) still gets the old version without that fix:

```
⊙  rm -rf venv && python3.8 -m venv venv; venv/bin/python -m pip install beets && rg -C3 'year =' venv/lib/python3.8/site-packages/beetsplug/spotify.py                                           julian@Air
Collecting beets
/Users/julian/Desktop/venv/lib/python3.8/site-packages/pip/_vendor/msgpack/fallback.py:133: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  unpacker = Unpacker(None, max_buffer_size=len(packed), **kwargs)
  Using cached https://files.pythonhosted.org/packages/3a/a8/6ac61e3ca16da6278c681381ed392bc02e6d410ea3ff3a4af34de1379b38/beets-1.4.9.tar.gz
Collecting six>=1.9 (from beets)
  Using cached https://files.pythonhosted.org/packages/ee/ff/48bde5c0f013094d729fe4b0316ba2a24774b3ff1c52d924a8a4cb04078a/six-1.15.0-py2.py3-none-any.whl
Collecting mutagen>=1.33 (from beets)
  Using cached https://files.pythonhosted.org/packages/49/d7/245d48972216d000238d6ccc09142bc1be88e707b38e8fb19a0155292cb6/mutagen-1.44.0-py3-none-any.whl
Collecting unidecode (from beets)
  Using cached https://files.pythonhosted.org/packages/d0/42/d9edfed04228bacea2d824904cae367ee9efd05e6cce7ceaaedd0b0ad964/Unidecode-1.1.1-py2.py3-none-any.whl
Collecting musicbrainzngs>=0.4 (from beets)
  Using cached https://files.pythonhosted.org/packages/6d/fd/cef7b2580436910ccd2f8d3deec0f3c81743e15c0eb5b97dde3fbf33c0c8/musicbrainzngs-0.7.1-py2.py3-none-any.whl
Collecting pyyaml (from beets)
  Using cached https://files.pythonhosted.org/packages/64/c2/b80047c7ac2478f9501676c988a5411ed5572f35d1beff9cae07d321512c/PyYAML-5.3.1.tar.gz
Collecting munkres>=1.0.0 (from beets)
  Using cached https://files.pythonhosted.org/packages/64/97/61ddc63578870e04db6eb1d3bee58ad4e727f682068a7c7405edb8b2cdeb/munkres-1.1.2-py2.py3-none-any.whl
Collecting jellyfish (from beets)
  Using cached https://files.pythonhosted.org/packages/04/3f/d03cb056f407ef181a45569255348457b1a0915fc4eb23daeceb930a68a4/jellyfish-0.8.2.tar.gz
Installing collected packages: six, mutagen, unidecode, musicbrainzngs, pyyaml, munkres, jellyfish, beets
  Running setup.py install for pyyaml ... done
  Running setup.py install for jellyfish ... done
  Running setup.py install for beets ... done
Successfully installed beets-1.4.9 jellyfish-0.8.2 munkres-1.1.2 musicbrainzngs-0.7.1 mutagen-1.44.0 pyyaml-5.3.1 six-1.15.0 unidecode-1.1.1
WARNING: You are using pip version 19.2.3, however version 20.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
173-            year, month = date_parts
174-            day = None
175-        elif release_date_precision == 'year':
176:            year = date_parts
177-            month = None
178-            day = None
179-        else:
```

**Comment by sampsyo** [comment]
Indeed; thanks! This was fixed in #3343 and we should have closed this issue. Release coming soon, hopefully…
